### DISCUSSION TITLE
ModuleNotFoundError: No module named 'movelladot_pc_sdk_py310_64'

### DISCUSSION THREAD

**Discussion by TristanARS** [pyinstaller/pyinstaller#8005] [Help] [answered]
Hello,
I'm trying to create an exe from my project python using the Movella SDK. I'm using pycharm and python 3.10. I have no issue regarding the import when running my script from pycharm. However, when I create my exe i have no errrors during the creation but when I try to run my exe from a terminal I have this error : ModuleNotFoundError: No module named 'movelladot_pc_sdk_py310_64'

I know this is a common issue and i tried many solutions that i found, adding -hidden-import to make sure all the modules are correctly added but this did not help. I don't know if it can be related to my venv but i made sure that pyinstaller is correctly installed by running :

(venv) C:\Users\User\PycharmProjects\pythonProjectDE>where pyinstaller                                                                                           
C:\Users\User\PycharmProjects\pythonProjectDE\venv\Scripts\pyinstaller.exe

I'm running out of ideas.. I hope someone can help me. Thank you

Tristan

**Comment by rokm** [comment, marked as answer]
That's because `movelladot_pc_sdk\__init__.py` looks like this:
```python
#!/usr/bin/env python
import os
import sys
here = os.path.dirname(os.path.realpath(__file__))
sys.path.append(here)
from movelladot_pc_sdk_py310_64 import *
```

and PyInstaller's PyiFrozenImporter does not support that sort of sys.path machinations. But this also prevents PyInstaller's import analysis from properly analyzing the `movelladot_pc_sdk_py310_64` and its dependencies.

Unfortunately the most straightforward way to deal with this is to fix that `__init__.py` file into:
```python
from .movelladot_pc_sdk_py310_64 import *
```
(note the dot before `movelladot_pc_sdk_py310_64` - this also makes the rest of the lines in there redundant).

**Reply by TristanARS** [reply]
Thank you for your answer this solved my issue ! I'm quite new to this, so I was wondering what does the dot changed to make this work instantly ?

**Reply by rokm** [reply]
By adding the dot, the `movelladot_pc_sdk_py310_64` becomes a proper submodule of `movelladot_pc_sdk`, and therefore becomes visible to PyInstaller's import analysis. It also makes package structure static (i.e., no dynamic path changing is required - the latter usually does not work well with PyInstaller). 

Whereas in the original `__init__.py`, the search path (`sys.path`) is manipulated so that `movelladot_pc_sdk_py310_64` becomes a top-level module. But since the search path modification is dynamic, PyInstaller's import analysis is unaware of it, and thus cannot find the faux-top-level module. You might also be able to work around this by adding package's directory to PyInstaller's search path via [`--paths`](https://pyinstaller.org/en/stable/usage.html#cmdoption-p). But it is cleaner to fix the package, since that path-modification is likely a remnant of support for old python versions.

**Reply by TristanARS** [reply]
Ok I see what you mean. Thank you very much, I will keep this in mind for the future !
